velox/substrait/SubstraitUtils.h

```diff
@@ -42,33 +42,43 @@ class SubstraitParser {
       const ::substrait::NamedStruct& namedStruct);
 
   /// Used to parse Substrait Type.
-  std::shared_ptr<SubstraitType> parseType(const ::substrait::Type& sType);
+  std::shared_ptr<SubstraitType> parseType(
+      const ::substrait::Type& substraitType);
 
   /// Used to parse Substrait ReferenceSegment.
   int32_t parseReferenceSegment(
-      const ::substrait::Expression::ReferenceSegment& sRef);
+      const ::substrait::Expression::ReferenceSegment& refSegment);
 
   /// Used to make names in the format of {prefix}_{index}.
   std::vector<std::string> makeNames(const std::string& prefix, int size);
 
   /// Used to make node name in the format of n{nodeId}_{colIdx}.
   std::string makeNodeName(int nodeId, int colIdx);
 
+  /// Used to get the column index from a node name in the format of
+  /// n{nodeId}_{colIdx}.
+  int getIdxFromNodeName(const std::string& nodeName);
+
   /// Used to find the Substrait function name according to the function id
   /// from a pre-constructed function map. The function specification can be
   /// a simple name or a compound name. The compound name format is:
   /// <function name>:<short_arg_type0>_<short_arg_type1>_..._<short_arg_typeN>.
   /// Currently, the input types in the function specification are not used. But
   /// in the future, they should be used for the validation according the
   /// specifications in Substrait yaml files.
-  std::string findSubstraitFuncSpec(
+  const std::string& findSubstraitFuncSpec(
       const std::unordered_map<uint64_t, std::string>& functionMap,
       uint64_t id) const;
 
   /// This function is used to get the function name from the compound name.
   /// When the input is a simple name, it will be returned.
   std::string getSubFunctionName(const std::string& subFuncSpec) const;
 
+  /// This function is used get the types from the compound name.
+  void getSubFunctionTypes(
```

Comment:
I keep reading "sub-function" as subscript function and not a Substrait function. Let's change Sub to Substrait or remove Sub if there is sufficient context already.

Also, for consistency:

findSubstraitFuncSpec, getSubFunctionTypes -> findFunctionSpec, getFunctionTypes (or extractFunctionTypes)

Reply:
Fixed.